### DISCUSSION TITLE
Map with Enum as Key in a Qute Template

### DISCUSSION THREAD

**Discussion by cdhermann** [quarkusio/quarkus#25780] [Q&A] [answered]
Dear Quarkus community,

does anybody know if and how it is possible in a Qute template to query a value from a Map when the key is an Enum?

Example:
```java
package a.b.c;

public enum Test {
    VAL1, VALS
}

var map = new HashMap<Test, String>();
```

In the Qute template neither `map.get(Test.VAL1)` nor `map.get(VAL1)` or `map.get(a.b.c.Test.VAL1)` is working.

At the moment I work around the issue by defining the HashMap as `new HashMap<String, String>`. In my opinion, this removes a lot of "compile time safety".

You help is apreciated.

[a comment by an automated account is not shown]

**Comment by mkouba** [comment, marked as answer]
You can annotate your enum with [`@TemplateEnum`](https://quarkus.io/guides/qute-reference#convenient-annotation-for-enums) and use something like `map.get(Test:VAL1)`. If you don't control the source of the enum you'd need to implement a template extension method like this: https://github.com/quarkusio/quarkus/blob/main/extensions/qute/deployment/src/test/java/io/quarkus/qute/deployment/extensions/NamespaceTemplateExtensionTest.java#L110-L113.

**Comment by cdhermann** [comment]
@mkouba thank you very much. It works.

It is also in the [docs](https://quarkus.io/guides/qute-reference#convenient-annotation-for-enums). But, until you answer here I lacked the brain power to relate this to my issue.
